draft-pda-protocol.md

```diff
@@ -950,31 +956,68 @@ indefinitely until a suitable output is found.
+are (1) the PDAParam structure configured out of band and (2) a corresponding value to be
+submitted in a report. Clients are not expected to store any state across any upload
+flows, nor are required to implement any sort of report retry mechanism should it fail.
+By design, the protocol in this document is robust against individual client upload
+failures since the protocol output is an aggregate over all inputs.
 
-[[OPEN ISSUE: writeme]]
+### Aggregator capabilities
 
-## Open questions and system parameters {#questions-and-params}
+Helpers and leaders have different operational requirements. The design in this
+document assumes an operationally competent leader, i.e., one that has no storage
+or computation limitations or constraints, and minimalistic helper, i.e., one that
+has computation, bandwidth, and storage constraints. By design, leaders must be
+at least as capable as helpers, where helpers are generally required to:
 
-[[OPEN ISSUE: discuss batch size parameter and thresholds]]
-[[OPEN ISSUE: discuss f^ leakage differences from [GB17]]]
+- Support the collect protocol, which includes verifying and aggregating
+  sets of reports in a given batch; and
+- Publish and manage an HPKE configuration that can be used for the upload protocol;
 
+In addition, for each PDAParam instance, helpers are required to:
 
-# Operational Considerations
+- Implement some form of batch-to-report and intra-batch replay mitigation storage,
+  which includes some way of tracking batch report size, with optional support for
+  state offloading.
 
-Prio has inherent constraints derived from the tradeoff between privacy
-guarantees and computational complexity. These tradeoffs influence how
-applications may choose to utilize services implementing the specification.
+[OPEN ISSUE: describe intra- and inter-batch replay attacks in the security considerations]
+
+Beyond the minimal capabiltiies required of helpers, leaders are generally required to:
+
+- Support the upload protocol and store client reports for a given PDAParam instance,
+  where each report maps uniquely to a single batch, and index this storage by batch durations;
+- Track batch report size during each collect flow and request encrypted output shares
+  from helpers.
+
+In addition, for each PDAParam instance, leaders are required to:
+
+- Ensure that, for each report, all aggregator shares are either included or not in a
+  given batch;
+- Implement and store state for some form of inter- and intra-batch replay mitigation; and
+- Store helper state associated with a given PDAParam batch.
+
+### Collector capabilities
+
+Collectors statefully interact with aggregators to produce an aggregate output. Their
+input to the protocol is the PAParam structure, configured out of band, which contains
+the corresponding batch window and size. For each collect invocation, collectors are
+required to keep state from the start of the protocol to the end as needed to produce
+the final aggregate output.
```

Comment:
They also need to keep around an HPKE secret key for the lifetime of the PDAParam.